routes/volunteering-events.js

```diff
@@ -21,7 +25,35 @@ function getRequestOrOffer(fieldName) {
   };
 }
 
+const getPlannedEvents = async (ctx) => {
+  try {
+    const requestedField = await VolunteeringEvent.find({ isDone: false })
+      .populate([
+        {
+          path: 'request',
+          model: VolunteeringRequest,
+          populate: [
+            { path: 'content', model: Content, select: ['title'] },
+            { path: 'regions', populate: { path: 'SubRegion', model: SubRegion } }],
+          select: ['title', 'contact', 'about', 'content', 'regions', 'organization'],
+        },
+        {
+          path:
+          'offer',
+          model: VolunteeringOffer,
+          populate: [
+            { path: 'content', model: Content, select: ['title'] },
+            { path: 'regions', populate: { path: 'SubRegion', model: SubRegion } }],
+          select: ['contact', 'content', 'regions', 'organization'],
+        }]);
+    DBMethods.handleDocResponse(ctx, requestedField);
+  } catch (error) {
+    DBMethods.handleErrors(ctx, error);
+  }
+};
+
 router.get('/', DBMethods.list(VolunteeringEvent))
+  .get('/planned', getPlannedEvents)
```

Comment:
This is not how REST API should look. It should be `/events?planned=true` or  `/events/?planned=true` (the same).
See more [here](https://github.com/paypal/api-standards/blob/master/api-style-guide.md#filter-a-resource-collection).

Reply:
Base-db-methods has to function. One is find which is not is used and the list function. The list function is used as the default get rest api. The list function seems to ignore query parameters which are not dates. 
Also the back-end does not support population of the ref fields out of the box which is much needed. 
So in order to move on and have something for the children I make custom API which is not RESTFULL API.

Reply:
I fixed the base-db-methods to support this.